store/httpstore.go

```diff
@@ -99,10 +99,6 @@ func (s HTTPStore) GetMeta(name string, size int64) ([]byte, error) {
 	logrus.Debugf("%d when retrieving metadata for %s", resp.StatusCode, name)
 	b := io.LimitReader(resp.Body, size)
 	body, err := ioutil.ReadAll(b)
-	if resp.ContentLength > 0 && int64(len(body)) < resp.ContentLength {
```

Comment:
If the data is invalid, yes, they checksum will catch it. This would potentially allow somebody to set a large size (in the snapshot/timestamp, as opposed to the client passing the 5MB max size for unknown sizes), send less data than the size indicates, but have the checksum be valid for that shorter object. As the checksum is a stronger validation, I think this should be treated as valid (if poor) behaviour.